Build failure: bitwarden-cli

### Steps To Reproduce

Steps to reproduce the behavior:
1. build bitwarden-cli on aarch64-darwin on unstable

### Build log

Relevant log lines:

```
330 npm error /nix/store/ap4xjkwp0lshyywxqw88phnnc0j1fnzw-bash-5.2p32/bin/sh: line 1: perl: command not found
331 npm error gyp: Call to 'cc -v 2>&1 | perl -ne 'print $1 if /clang version ([0-9]+(\.[0-9]+){2,})/'' returned exit sta>
332 npm error gyp ERR! configure error
333 npm error gyp ERR! stack Error: `gyp` failed with exit code: 1
334 npm error gyp ERR! stack     at ChildProcess.onCpExit (/private/tmp/nix-build-bitwarden-cli-2024.8.1.drv-3/source/nod>
335 npm error gyp ERR! stack     at ChildProcess.emit (node:events:519:28)
336 npm error gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:294:12)
337 npm error gyp ERR! System Darwin 23.6.0
338 npm error gyp ERR! command "/nix/store/iikbj8fwd3msxvh747v22b0gpf1rckww-nodejs-20.16.0/bin/node" "/private/tmp/nix-bu>
339 npm error gyp ERR! cwd /private/tmp/nix-build-bitwarden-cli-2024.8.1.drv-3/source/node_modules/bufferutil
340 npm error gyp ERR! node -v v20.16.0
341 npm error gyp ERR! node-gyp -v v9.4.1
342 npm error gyp ERR! not ok
```

### Notify maintainers

@dotlambda 

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
this path will be fetched (0.00 MiB download, 0.00 MiB unpacked):
  /nix/store/3bj2c590msmkn6qpc3q3z5jayki83afn-nix-info
copying path '/nix/store/3bj2c590msmkn6qpc3q3z5jayki83afn-nix-info' from 'https://cache.nixos.org'...
 - system: `"aarch64-darwin"`
 - host os: `Darwin 23.6.0, macOS 14.6.1`
 - multi-user?: `yes`
 - sandbox: `no`
 - version: `nix-env (Nix) 2.18.5`
 - channels(root): `"nixpkgs"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixpkgs`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

> Anyone who gets stuck with an error similar to: `error: implicit instantiation of undefined template 'std::char_traits<unsigned char>'`
> 
> This is caused by libcxx removing the std::char_traits template in v19 (default used by stdenv)
> 
> The simple fix is to create an overlay to use llvm 18 and cc 18: `nixpkgs.overlays = [ (final: prev: { bitwarden-cli = prev.bitwarden-cli.overrideAttrs (oldAttrs: { nativeBuildInputs = (oldAttrs.nativeBuildInputs or [ ]) ++ [ prev.llvmPackages_18.stdenv.cc ]; stdenv = prev.llvmPackages_18.stdenv; }); }) ];`

That did not solve it for me unfortunately. Still links/uses `libcxx-19.1.5-dev/`

```
error: builder for '/nix/store/66j2p8qqkrqk5w7b453sy6d2s0cdqgg3-bitwarden-cli-2024.12.0.drv' failed with exit code 1;
       last 25 log lines:
       > npm error       |     ^
       > npm error /nix/store/0nhi47d5ip48wprxnava6vv973zzzndf-libcxx-19.1.5-dev/include/c++/v1/string:3304:5: note: in instantiation of member function 'std::basic_string<unsigned char>::__erase_to_end' requested here
       > npm error  3304 |     __erase_to_end(__n);
       > npm error       |     ^
       > npm error /nix/store/0nhi47d5ip48wprxnava6vv973zzzndf-libcxx-19.1.5-dev/include/c++/v1/string:1303:84: note: in instantiation of member function 'std::basic_string<unsigned char>::resize' requested here
       > npm error  1303 |   _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 void resize(size_type __n) { resize(__n, value_type()); }
       > npm error       |                                                                                    ^
       > npm error ../argon2.cpp:29:14: note: in instantiation of member function 'std::basic_string<unsigned char>::resize' requested here
       > npm error    29 |         hash.resize(hash_length);
       > npm error       |              ^
       > npm error /nix/store/0nhi47d5ip48wprxnava6vv973zzzndf-libcxx-19.1.5-dev/include/c++/v1/__fwd/string.h:23:29: note: template is declared here
       > npm error    23 | struct _LIBCPP_TEMPLATE_VIS char_traits;
       > npm error       |                             ^
       > npm error 9 errors generated.
       > npm error make: *** [argon2.target.mk:138: Release/obj.target/argon2/argon2.o] Error 1
       > npm error gyp ERR! build error
       > npm error gyp ERR! stack Error: `make` failed with exit code: 2
       > npm error gyp ERR! stack at ChildProcess.<anonymous> (/private/tmp/nix-build-bitwarden-cli-2024.12.0.drv-0/source/node_modules/node-gyp/lib/build.js:216:23)
       > npm error gyp ERR! System Darwin 24.2.0
       > npm error gyp ERR! command "/nix/store/q49zbi0pzjhg4zn085h9hyz9m1k3hvpb-nodejs-20.18.1/bin/node" "/private/tmp/nix-build-bitwarden-cli-2024.12.0.drv-0/source/node_modules/node-gyp/bin/node-gyp.js" "rebuild"
       > npm error gyp ERR! cwd /private/tmp/nix-build-bitwarden-cli-2024.12.0.drv-0/source/node_modules/argon2
       > npm error gyp ERR! node -v v20.18.1
       > npm error gyp ERR! node-gyp -v v10.2.0
       > npm error gyp ERR! not ok
       > npm error A complete log of this run can be found in: /private/tmp/nix-build-bitwarden-cli-2024.12.0.drv-0/cache/_logs/2025-01-13T13_49_10_822Z-debug-0.log
       For full logs, run 'nix log /nix/store/66j2p8qqkrqk5w7b453sy6d2s0cdqgg3-bitwarden-cli-2024.12.0.drv'.
```

--- a/pkgs/by-name/bi/bitwarden-cli/fix-lockfile.patch
+++ b/pkgs/by-name/bi/bitwarden-cli/fix-lockfile.patch
@@ -1,0 +1,567 @@
+diff --git a/package-lock.json b/package-lock.json
+index 1b4fe5f40e..fe3166797e 100644
+--- a/package-lock.json
++++ b/package-lock.json
+@@ -230,51 +230,6 @@
+         "bw": "build/bw.js"
+       }
+     },
+-    "apps/cli/node_modules/define-lazy-prop": {
+-      "version": "2.0.0",
+-      "license": "MIT",
+-      "engines": {
+-        "node": ">=8"
+-      }
+-    },
+-    "apps/cli/node_modules/is-docker": {
+-      "version": "2.2.1",
+-      "license": "MIT",
+-      "bin": {
+-        "is-docker": "cli.js"
+-      },
+-      "engines": {
+-        "node": ">=8"
+-      },
+-      "funding": {
+-        "url": "https://github.com/sponsors/sindresorhus"
+-      }
+-    },
+-    "apps/cli/node_modules/is-wsl": {
+-      "version": "2.2.0",
+-      "license": "MIT",
+-      "dependencies": {
+-        "is-docker": "^2.0.0"
+-      },
+-      "engines": {
+-        "node": ">=8"
+-      }
+-    },
+-    "apps/cli/node_modules/open": {
+-      "version": "8.4.2",
+-      "license": "MIT",
+-      "dependencies": {
+-        "define-lazy-prop": "^2.0.0",
+-        "is-docker": "^2.1.1",
+-        "is-wsl": "^2.2.0"
+-      },
+-      "engines": {
+-        "node": ">=12"
+-      },
+-      "funding": {
+-        "url": "https://github.com/sponsors/sindresorhus"
+-      }
+-    },
+     "apps/desktop": {
+       "name": "@bitwarden/desktop",
+       "version": "2025.9.0",
+@@ -4913,33 +4868,6 @@
+         }
+       }
+     },
+-    "node_modules/@compodoc/compodoc/node_modules/@angular-devkit/schematics/node_modules/chokidar": {
+-      "version": "3.6.0",
+-      "resolved": "https://registry.npmjs.org/chokidar/-/chokidar-3.6.0.tgz",
+-      "integrity": "sha512-7VT13fmjotKpGipCW9JEQAusEPE+Ei8nl6/g4FBAmIm0GOOLMua9NDDo/DWp0ZAxCr3cPq5ZpBqmPAQgDda2Pw==",
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true,
+-      "dependencies": {
+-        "anymatch": "~3.1.2",
+-        "braces": "~3.0.2",
+-        "glob-parent": "~5.1.2",
+-        "is-binary-path": "~2.1.0",
+-        "is-glob": "~4.0.1",
+-        "normalize-path": "~3.0.0",
+-        "readdirp": "~3.6.0"
+-      },
+-      "engines": {
+-        "node": ">= 8.10.0"
+-      },
+-      "funding": {
+-        "url": "https://paulmillr.com/funding/"
+-      },
+-      "optionalDependencies": {
+-        "fsevents": "~2.3.2"
+-      }
+-    },
+     "node_modules/@compodoc/compodoc/node_modules/@babel/core": {
+       "version": "7.25.8",
+       "resolved": "https://registry.npmjs.org/@babel/core/-/core-7.25.8.tgz",
+@@ -5122,21 +5050,6 @@
+       "dev": true,
+       "license": "MIT"
+     },
+-    "node_modules/@compodoc/compodoc/node_modules/glob-parent": {
+-      "version": "5.1.2",
+-      "resolved": "https://registry.npmjs.org/glob-parent/-/glob-parent-5.1.2.tgz",
+-      "integrity": "sha512-AOIgSQCepiJYwP3ARnGx+5VnTu2HBYdzbGP45eLw1vr3zB3vZLeyed1sC9hnbcOc9/SrMyM5RPQrkGz4aS9Zow==",
+-      "dev": true,
+-      "license": "ISC",
+-      "optional": true,
+-      "peer": true,
+-      "dependencies": {
+-        "is-glob": "^4.0.1"
+-      },
+-      "engines": {
+-        "node": ">= 6"
+-      }
+-    },
+     "node_modules/@compodoc/compodoc/node_modules/magic-string": {
+       "version": "0.30.11",
+       "resolved": "https://registry.npmjs.org/magic-string/-/magic-string-0.30.11.tgz",
+@@ -5147,36 +5060,6 @@
+         "@jridgewell/sourcemap-codec": "^1.5.0"
+       }
+     },
+-    "node_modules/@compodoc/compodoc/node_modules/readdirp": {
+-      "version": "3.6.0",
+-      "resolved": "https://registry.npmjs.org/readdirp/-/readdirp-3.6.0.tgz",
+-      "integrity": "sha512-hOS089on8RduqdbhvQ5Z37A0ESjsqz6qnRcffsMU3495FuTdqSm+7bhJ29JvIOsBDEEnan5DPu9t3To9VRlMzA==",
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true,
+-      "dependencies": {
+-        "picomatch": "^2.2.1"
+-      },
+-      "engines": {
+-        "node": ">=8.10.0"
+-      }
+-    },
+-    "node_modules/@compodoc/compodoc/node_modules/readdirp/node_modules/picomatch": {
+-      "version": "2.3.1",
+-      "resolved": "https://registry.npmjs.org/picomatch/-/picomatch-2.3.1.tgz",
+-      "integrity": "sha512-JU3teHTNjmE2VCGFzuY8EXzCDVwEqB2a8fsIvwaStHhAWJEeVd1o1QD80CU6+ZdEXXSLbSsuLwJjkCBWqRQUVA==",
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true,
+-      "engines": {
+-        "node": ">=8.6"
+-      },
+-      "funding": {
+-        "url": "https://github.com/sponsors/jonschlinkert"
+-      }
+-    },
+     "node_modules/@compodoc/live-server": {
+       "version": "1.2.3",
+       "resolved": "https://registry.npmjs.org/@compodoc/live-server/-/live-server-1.2.3.tgz",
+@@ -5908,28 +5791,6 @@
+         "node": ">=16.4"
+       }
+     },
+-    "node_modules/@electron/windows-sign": {
+-      "version": "1.2.2",
+-      "resolved": "https://registry.npmjs.org/@electron/windows-sign/-/windows-sign-1.2.2.tgz",
+-      "integrity": "sha512-dfZeox66AvdPtb2lD8OsIIQh12Tp0GNCRUDfBHIKGpbmopZto2/A8nSpYYLoedPIHpqkeblZ/k8OV0Gy7PYuyQ==",
+-      "dev": true,
+-      "license": "BSD-2-Clause",
+-      "optional": true,
+-      "peer": true,
+-      "dependencies": {
+-        "cross-dirname": "^0.1.0",
+-        "debug": "^4.3.4",
+-        "fs-extra": "^11.1.1",
+-        "minimist": "^1.2.8",
+-        "postject": "^1.0.0-alpha.6"
+-      },
+-      "bin": {
+-        "electron-windows-sign": "bin/electron-windows-sign.js"
+-      },
+-      "engines": {
+-        "node": ">=14.14"
+-      }
+-    },
+     "node_modules/@emnapi/core": {
+       "version": "1.4.3",
+       "resolved": "https://registry.npmjs.org/@emnapi/core/-/core-1.4.3.tgz",
+@@ -8285,21 +8146,6 @@
+         "@msgpack/msgpack": "^2.7.0"
+       }
+     },
+-    "node_modules/@microsoft/signalr/node_modules/utf-8-validate": {
+-      "version": "5.0.10",
+-      "resolved": "https://registry.npmjs.org/utf-8-validate/-/utf-8-validate-5.0.10.tgz",
+-      "integrity": "sha512-Z6czzLq4u8fPOyx7TU6X3dvUZVvoJmxSQ+IcrlmagKhilxlhZgxPK6C5Jqbkw1IDUmFTM+cz9QDnnLTwDz/2gQ==",
+-      "hasInstallScript": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true,
+-      "dependencies": {
+-        "node-gyp-build": "^4.3.0"
+-      },
+-      "engines": {
+-        "node": ">=6.14.2"
+-      }
+-    },
+     "node_modules/@microsoft/signalr/node_modules/ws": {
+       "version": "7.5.10",
+       "resolved": "https://registry.npmjs.org/ws/-/ws-7.5.10.tgz",
+@@ -11310,21 +11156,6 @@
+         "linux"
+       ]
+     },
+-    "node_modules/@rollup/rollup-linux-riscv64-musl": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-riscv64-musl/-/rollup-linux-riscv64-musl-4.44.0.tgz",
+-      "integrity": "sha512-TDKO8KlHJuvTEdfw5YYFBjhFts2TR0VpZsnLLSYmB7AaohJhM8ctDSdDnUGq77hUh4m/djRafw+9zQpkOanE2Q==",
+-      "cpu": [
+-        "riscv64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+     "node_modules/@rollup/rollup-linux-s390x-gnu": {
+       "version": "4.34.8",
+       "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-s390x-gnu/-/rollup-linux-s390x-gnu-4.34.8.tgz",
+@@ -18997,15 +18828,6 @@
+       "integrity": "sha512-vQOuWmBgsgG1ovGeDi8m6Zeu1JaqH/JncrxKmaqMbv/LunyOQdLiQhPHtOsNlbUI05TocR5nod/Mbs3HYtr6sQ==",
+       "license": "MIT"
+     },
+-    "node_modules/cross-dirname": {
+-      "version": "0.1.0",
+-      "resolved": "https://registry.npmjs.org/cross-dirname/-/cross-dirname-0.1.0.tgz",
+-      "integrity": "sha512-+R08/oI0nl3vfPcqftZRpytksBXDzOUveBq/NBVx0sUp1axwzPQrKinNx5yd5sxPu8j1wIy8AfnVQ+5eFdha6Q==",
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true
+-    },
+     "node_modules/cross-env": {
+       "version": "10.0.0",
+       "resolved": "https://registry.npmjs.org/cross-env/-/cross-env-10.0.0.tgz",
+@@ -33661,36 +33483,6 @@
+       "dev": true,
+       "license": "MIT"
+     },
+-    "node_modules/postject": {
+-      "version": "1.0.0-alpha.6",
+-      "resolved": "https://registry.npmjs.org/postject/-/postject-1.0.0-alpha.6.tgz",
+-      "integrity": "sha512-b9Eb8h2eVqNE8edvKdwqkrY6O7kAwmI8kcnBv1NScolYJbo59XUF0noFq+lxbC1yN20bmC0WBEbDC5H/7ASb0A==",
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true,
+-      "dependencies": {
+-        "commander": "^9.4.0"
+-      },
+-      "bin": {
+-        "postject": "dist/cli.js"
+-      },
+-      "engines": {
+-        "node": ">=14.0.0"
+-      }
+-    },
+-    "node_modules/postject/node_modules/commander": {
+-      "version": "9.5.0",
+-      "resolved": "https://registry.npmjs.org/commander/-/commander-9.5.0.tgz",
+-      "integrity": "sha512-KRs7WVDKg86PWiuAqhDrAQnTXZKraVcCc6vFdL14qrZ/DcWwuRo7VoiYXalXO7S5GKpqYiVEwCbgFDfxNHKJBQ==",
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "peer": true,
+-      "engines": {
+-        "node": "^12.20.0 || >=14"
+-      }
+-    },
+     "node_modules/prebuild-install": {
+       "version": "7.1.3",
+       "resolved": "https://registry.npmjs.org/prebuild-install/-/prebuild-install-7.1.3.tgz",
+@@ -39330,291 +39122,6 @@
+         }
+       }
+     },
+-    "node_modules/vite/node_modules/@rollup/rollup-android-arm-eabi": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-android-arm-eabi/-/rollup-android-arm-eabi-4.44.0.tgz",
+-      "integrity": "sha512-xEiEE5oDW6tK4jXCAyliuntGR+amEMO7HLtdSshVuhFnKTYoeYMyXQK7pLouAJJj5KHdwdn87bfHAR2nSdNAUA==",
+-      "cpu": [
+-        "arm"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "android"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-android-arm64": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-android-arm64/-/rollup-android-arm64-4.44.0.tgz",
+-      "integrity": "sha512-uNSk/TgvMbskcHxXYHzqwiyBlJ/lGcv8DaUfcnNwict8ba9GTTNxfn3/FAoFZYgkaXXAdrAA+SLyKplyi349Jw==",
+-      "cpu": [
+-        "arm64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "android"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-darwin-arm64": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-darwin-arm64/-/rollup-darwin-arm64-4.44.0.tgz",
+-      "integrity": "sha512-VGF3wy0Eq1gcEIkSCr8Ke03CWT+Pm2yveKLaDvq51pPpZza3JX/ClxXOCmTYYq3us5MvEuNRTaeyFThCKRQhOA==",
+-      "cpu": [
+-        "arm64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "darwin"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-darwin-x64": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-darwin-x64/-/rollup-darwin-x64-4.44.0.tgz",
+-      "integrity": "sha512-fBkyrDhwquRvrTxSGH/qqt3/T0w5Rg0L7ZIDypvBPc1/gzjJle6acCpZ36blwuwcKD/u6oCE/sRWlUAcxLWQbQ==",
+-      "cpu": [
+-        "x64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "darwin"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-freebsd-arm64": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-freebsd-arm64/-/rollup-freebsd-arm64-4.44.0.tgz",
+-      "integrity": "sha512-u5AZzdQJYJXByB8giQ+r4VyfZP+walV+xHWdaFx/1VxsOn6eWJhK2Vl2eElvDJFKQBo/hcYIBg/jaKS8ZmKeNQ==",
+-      "cpu": [
+-        "arm64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "freebsd"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-freebsd-x64": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-freebsd-x64/-/rollup-freebsd-x64-4.44.0.tgz",
+-      "integrity": "sha512-qC0kS48c/s3EtdArkimctY7h3nHicQeEUdjJzYVJYR3ct3kWSafmn6jkNCA8InbUdge6PVx6keqjk5lVGJf99g==",
+-      "cpu": [
+-        "x64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "freebsd"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-arm-gnueabihf": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-arm-gnueabihf/-/rollup-linux-arm-gnueabihf-4.44.0.tgz",
+-      "integrity": "sha512-x+e/Z9H0RAWckn4V2OZZl6EmV0L2diuX3QB0uM1r6BvhUIv6xBPL5mrAX2E3e8N8rEHVPwFfz/ETUbV4oW9+lQ==",
+-      "cpu": [
+-        "arm"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-arm-musleabihf": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-arm-musleabihf/-/rollup-linux-arm-musleabihf-4.44.0.tgz",
+-      "integrity": "sha512-1exwiBFf4PU/8HvI8s80icyCcnAIB86MCBdst51fwFmH5dyeoWVPVgmQPcKrMtBQ0W5pAs7jBCWuRXgEpRzSCg==",
+-      "cpu": [
+-        "arm"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-arm64-gnu": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-arm64-gnu/-/rollup-linux-arm64-gnu-4.44.0.tgz",
+-      "integrity": "sha512-ZTR2mxBHb4tK4wGf9b8SYg0Y6KQPjGpR4UWwTFdnmjB4qRtoATZ5dWn3KsDwGa5Z2ZBOE7K52L36J9LueKBdOQ==",
+-      "cpu": [
+-        "arm64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-arm64-musl": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-arm64-musl/-/rollup-linux-arm64-musl-4.44.0.tgz",
+-      "integrity": "sha512-GFWfAhVhWGd4r6UxmnKRTBwP1qmModHtd5gkraeW2G490BpFOZkFtem8yuX2NyafIP/mGpRJgTJ2PwohQkUY/Q==",
+-      "cpu": [
+-        "arm64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-loongarch64-gnu": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-loongarch64-gnu/-/rollup-linux-loongarch64-gnu-4.44.0.tgz",
+-      "integrity": "sha512-xw+FTGcov/ejdusVOqKgMGW3c4+AgqrfvzWEVXcNP6zq2ue+lsYUgJ+5Rtn/OTJf7e2CbgTFvzLW2j0YAtj0Gg==",
+-      "cpu": [
+-        "loong64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-powerpc64le-gnu": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-powerpc64le-gnu/-/rollup-linux-powerpc64le-gnu-4.44.0.tgz",
+-      "integrity": "sha512-bKGibTr9IdF0zr21kMvkZT4K6NV+jjRnBoVMt2uNMG0BYWm3qOVmYnXKzx7UhwrviKnmK46IKMByMgvpdQlyJQ==",
+-      "cpu": [
+-        "ppc64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-riscv64-gnu": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-riscv64-gnu/-/rollup-linux-riscv64-gnu-4.44.0.tgz",
+-      "integrity": "sha512-vV3cL48U5kDaKZtXrti12YRa7TyxgKAIDoYdqSIOMOFBXqFj2XbChHAtXquEn2+n78ciFgr4KIqEbydEGPxXgA==",
+-      "cpu": [
+-        "riscv64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-s390x-gnu": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-s390x-gnu/-/rollup-linux-s390x-gnu-4.44.0.tgz",
+-      "integrity": "sha512-8541GEyktXaw4lvnGp9m84KENcxInhAt6vPWJ9RodsB/iGjHoMB2Pp5MVBCiKIRxrxzJhGCxmNzdu+oDQ7kwRA==",
+-      "cpu": [
+-        "s390x"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-x64-gnu": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-x64-gnu/-/rollup-linux-x64-gnu-4.44.0.tgz",
+-      "integrity": "sha512-iUVJc3c0o8l9Sa/qlDL2Z9UP92UZZW1+EmQ4xfjTc1akr0iUFZNfxrXJ/R1T90h/ILm9iXEY6+iPrmYB3pXKjw==",
+-      "cpu": [
+-        "x64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-linux-x64-musl": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-linux-x64-musl/-/rollup-linux-x64-musl-4.44.0.tgz",
+-      "integrity": "sha512-PQUobbhLTQT5yz/SPg116VJBgz+XOtXt8D1ck+sfJJhuEsMj2jSej5yTdp8CvWBSceu+WW+ibVL6dm0ptG5fcA==",
+-      "cpu": [
+-        "x64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "linux"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-win32-arm64-msvc": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-win32-arm64-msvc/-/rollup-win32-arm64-msvc-4.44.0.tgz",
+-      "integrity": "sha512-M0CpcHf8TWn+4oTxJfh7LQuTuaYeXGbk0eageVjQCKzYLsajWS/lFC94qlRqOlyC2KvRT90ZrfXULYmukeIy7w==",
+-      "cpu": [
+-        "arm64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "win32"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-win32-ia32-msvc": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-win32-ia32-msvc/-/rollup-win32-ia32-msvc-4.44.0.tgz",
+-      "integrity": "sha512-3XJ0NQtMAXTWFW8FqZKcw3gOQwBtVWP/u8TpHP3CRPXD7Pd6s8lLdH3sHWh8vqKCyyiI8xW5ltJScQmBU9j7WA==",
+-      "cpu": [
+-        "ia32"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "win32"
+-      ],
+-      "peer": true
+-    },
+-    "node_modules/vite/node_modules/@rollup/rollup-win32-x64-msvc": {
+-      "version": "4.44.0",
+-      "resolved": "https://registry.npmjs.org/@rollup/rollup-win32-x64-msvc/-/rollup-win32-x64-msvc-4.44.0.tgz",
+-      "integrity": "sha512-Q2Mgwt+D8hd5FIPUuPDsvPR7Bguza6yTkJxspDGkZj7tBRn2y4KSWYuIXpftFSjBra76TbKerCV7rgFPQrn+wQ==",
+-      "cpu": [
+-        "x64"
+-      ],
+-      "dev": true,
+-      "license": "MIT",
+-      "optional": true,
+-      "os": [
+-        "win32"
+-      ],
+-      "peer": true
+-    },
+     "node_modules/vite/node_modules/rollup": {
+       "version": "4.44.0",
+       "resolved": "https://registry.npmjs.org/rollup/-/rollup-4.44.0.tgz",

--- a/pkgs/by-name/bi/bitwarden-cli/package.nix
+++ b/pkgs/by-name/bi/bitwarden-cli/package.nix
@@ -104,6 +104,9 @@ buildNpmPackage rec {
     homepage = "https://bitwarden.com";
     license = lib.licenses.gpl3Only;
     mainProgram = "bw";
-    maintainers = with lib.maintainers; [ dotlambda ];
+    maintainers = with lib.maintainers; [
+      xiaoxiangmoe
+      dotlambda
+    ];
   };
 }

--- a/pkgs/by-name/bi/bitwarden-cli/package.nix
+++ b/pkgs/by-name/bi/bitwarden-cli/package.nix
@@ -2,37 +2,41 @@
   lib,
   stdenv,
   buildNpmPackage,
-  nodejs_20,
   fetchFromGitHub,
-  cctools,
-  nix-update-script,
-  nixosTests,
+  nodejs_22,
   perl,
   xcbuild,
+  writableTmpDirAsHomeHook,
+  versionCheckHook,
+  nixosTests,
+  nix-update-script,
 }:
 
-buildNpmPackage rec {
+buildNpmPackage (finalAttrs: {
   pname = "bitwarden-cli";
-  version = "2025.5.0";
+  version = "2025.9.0";
 
   src = fetchFromGitHub {
     owner = "bitwarden";
     repo = "clients";
-    tag = "cli-v${version}";
-    hash = "sha256-8jVKwqKhTfhur226SER4sb1i4dY+TjJRYmOY8YtO6CY=";
+    tag = "cli-v${finalAttrs.version}";
+    hash = "sha256-vxGyDYtv0O5U4pnVrQm/BOIpDtpcDUOyFFdBDehQ2to=";
   };
+
+  patches = [
+    ./fix-lockfile.patch
+  ];
 
   postPatch = ''
     # remove code under unfree license
     rm -r bitwarden_license
   '';
 
-  nodejs = nodejs_20;
+  nodejs = nodejs_22;
 
-  npmDepsHash = "sha256-0IoBPRGdtkMeTrT5cqZLHB/WrUCONtsJ6YHh0y4K5Ls=";
+  npmDepsHash = "sha256-bn39QlZXNUa/GEZhJsjLiG3PRYdQ/Y36Tvef2fXH8yQ=";
 
   nativeBuildInputs = lib.optionals stdenv.hostPlatform.isDarwin [
-    cctools
     perl
     xcbuild.xcrun
   ];
@@ -61,9 +65,6 @@ buildNpmPackage rec {
     rm -r node_modules/**/prebuilds
     shopt -u globstar
 
-    # FIXME one of the esbuild versions fails to download @esbuild/linux-x64
-    rm -r node_modules/esbuild node_modules/vite/node_modules/esbuild
-
     npm rebuild --verbose
   '';
 
@@ -84,6 +85,14 @@ buildNpmPackage rec {
     installShellCompletion --cmd bw --zsh <($out/bin/bw completion --shell zsh)
   '';
 
+  doInstallCheck = true;
+  nativeInstallCheckInputs = [
+    writableTmpDirAsHomeHook
+    versionCheckHook
+  ];
+  versionCheckKeepEnvironment = [ "HOME" ];
+  versionCheckProgramArg = "--version";
+
   passthru = {
     tests = {
       vaultwarden = nixosTests.vaultwarden.sqlite;
@@ -97,9 +106,7 @@ buildNpmPackage rec {
   };
 
   meta = {
-    # https://github.com/NixOS/nixpkgs/issues/339576
-    broken = stdenv.hostPlatform.isDarwin;
-    changelog = "https://github.com/bitwarden/clients/releases/tag/${src.tag}";
+    changelog = "https://github.com/bitwarden/clients/releases/tag/${finalAttrs.src.tag}";
     description = "Secure and free password manager for all of your devices";
     homepage = "https://bitwarden.com";
     license = lib.licenses.gpl3Only;
@@ -109,4 +116,4 @@ buildNpmPackage rec {
       dotlambda
     ];
   };
-}
+})

